docs/cryptography/vigenere-cipher.mdx

```diff
@@ -0,0 +1,230 @@
+using namespace std;
+
+string encryption(string message, int x, string key) {
+	string cipher;
+	for (int i = 0; i < message.size(); ++i) {
+		int t=(message[i] + key[i%x])%26;
+		cipher+=('A'+t);
+	}
+	return cipher;
+}
+
+string decryption(string cipher, int x, string key) {
+	string message;
+	for (int i = 0; i < cipher.size(); ++i) {
+		int t=(cipher[i] - key[i%x] +26)%26;
+		message+=('A'+t);
+	}
+	return message;
+}
+
+int main() {
+
+	//use only uppercase character
+	int choice;
+	cout << "Choose 1 if you want to ENCRYPT or 2 if you want to DECRYPT: ";
+	cin >> choice;
+
+	if(choice == 1) {
+		string message;
+		cout << "Enter a plaintext message: ";
+		cin >> message;
+
+		string key;
+		cout << "Enter a keyword: ";
+		cin >> key;
+
+		cout << "\nEncrypted message: " << encryption(message, key.size(), key);
+		cout << endl;
+	}
+
+	else if(choice == 2) {
+		string ciphertext;
+		cout << "Enter a ciphertext: ";
+		cin >> ciphertext;
+
+		string key;
+		cout << "Enter a keyword: ";
+		cin >> key;
+
+		cout << "\nDecrypted message: " << decryption(ciphertext, key.size(), key);
+		cout << endl;
+	}
+
+	else {
+		cout << "\nInvalid choice." << endl;
+	}
+	return 0;
+}
+```
+
+</TabItem>
+<TabItem value="py">
+
+```python
+def encrypt(msg, key):
+    cipher = ""
+    n = len(key)
+    i = 0
+    for l in msg:
+        # finding the index of the character
+        p = int(ord(l) + ord(key[i % n])) % 26
+        # to perform the shift with the key
+        cipher += chr(p + ord('A'))
+        i += 1
+    return cipher
+
+
+def decrypt(cipher, key):
+    msg = ""
+    n = len(key)
+    j = 0
+    for l in cipher:
+        p = int(ord(l) - ord(key[j % n]) + 26) % 26
+        msg += chr(p + ord('A'))
+        j += 1
+    return msg
+
+if __name__=='__main__':
+    
+    # Use uppercase only
+    choice = int(input("Choose 1 if you want to ENCRYPT or 2 if you want to DECRYPT: "))
+
+    if(choice == 1):
+        msg = input("Enter a plaintext message: ")
+        key = input("Enter a keyword: ")
+
+        # encrypting the message
+        cipher = encrypt(msg, key)
+        print("\nEncrypted message: ", cipher)
+
+    elif(choice == 2):
+        cipher = input("Enter a ciphertext: ")
+        key = input("Enter a keyword: ")
+
+        # decrypting the ciphar text
+        original = decrypt(cipher, key)
+        print("\nDecrypted message: ", original)
+
+    else:
+        print("\nInvalid choice.")
+```
+
+</TabItem>
+</Tabs>
+
+## Sample Input and Output
+
+#### Input
+
+```
+Choose 1 if you want to ENCRYPT or 2 if you want to DECRYPT: 1
+Enter a plaintext message: NEOALGO
+Enter a keyword: MARS
+```
+
+#### Output
+
+```
+Encrypted message:  ZEFSXGF
+```
+
+#### Input
+
+```
+Choose 1 if you want to ENCRYPT or 2 if you want to DECRYPT: 2
+Enter a ciphertext: ZEFSXGF
+Enter a keyword: MARS
+```
+
+#### Output
+
+```
+Decrypted message: NEOALGO
+```
+
+## Complexity Analysis
+
+```
+Time Complexity - O(n)
+Space Complexity - O(1)
+```
```

Comment:
No need of keeping them in triple backticks. Use bullet points and keep the value in backticks